### PR TITLE
Add Cpu Arch in local logs and telemetry events

### PR DESCRIPTION
**Pull request by maddieford**
## Description
This PR adds CPU Arch to local logs during service initialization. It also adds CPU Arch to the 'KeywordName' column as a common parameter in our telemetry events.

Local logs:
2023-10-05T02:07:24.787099Z INFO ExtHandler ExtHandler Azure Linux Agent (Goal State Agent version 9.9.9.9)
2023-10-05T02:07:24.787282Z INFO ExtHandler ExtHandler OS: ubuntu 20.04
2023-10-05T02:07:24.787379Z INFO ExtHandler ExtHandler Python: 3.8.10
2023-10-05T02:07:24.787477Z INFO ExtHandler ExtHandler **CPU Arch: x86_64**
2023-10-05T02:07:24.795884Z INFO ExtHandler ExtHandler Distro: ubuntu-20.04; OSUtil: Ubuntu18OSUtil; AgentService: walinuxagent; Python: 3.8.10; **Arch: x86_64**; systemd: True; LISDrivers: name:           hv_vmbus
; logrotate: logrotate 3.14.0;

Telemetry:
OpcodeName | Name | KeywordName | Operation | OperationSuccess | Message
-- | -- | -- | -- | -- | --
2023-10-05T02:07:25.167405Z | WALinuxAgent | {"CpuArchitecture": "x86_64"} | OSInfo | true | Distro: ubuntu-20.04; OSUtil: Ubuntu18OSUtil; AgentService: walinuxagent; Python: 3.8.10; Arch: x86_64; systemd: True; LISDrivers: name:           hv_vmbus ; logrotate: logrotate 3.14.0;

</div>

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/event.py [hunk 1]
```diff
@@ -383,6 +383,10 @@ def __init__(self):
         self._common_parameters.append(TelemetryEventParam(CommonTelemetryEventSchema.VMId, "VMId_UNINITIALIZED"))
         self._common_parameters.append(TelemetryEventParam(CommonTelemetryEventSchema.ImageOrigin, 0))
 
+    @staticmethod
+    def _get_vm_arch():
+        return platform.machine()
```
Comment (narrieta) [inline azurelinuxagent/common/event.py:388]: can this ever raise an exception? should we catch and return 'unknown' on failure?

File: azurelinuxagent/ga/update.py [hunk 2]
```diff
@@ -315,18 +315,22 @@ def run(self, debug=False):
             logger.info("OS: {0} {1}", DISTRO_NAME, DISTRO_VERSION)
             logger.info("Python: {0}.{1}.{2}", PY_VERSION_MAJOR, PY_VERSION_MINOR, PY_VERSION_MICRO)
 
+            vm_arch = self._get_vm_arch()
+            logger.info("CPU Arch: {0}", vm_arch)
```
Comment (narrieta) [inline azurelinuxagent/ga/update.py:319]: do we need to log here? we are already logging below

File: azurelinuxagent/ga/update.py [hunk 3]
```diff
@@ -1013,13 +1017,10 @@ def _send_heartbeat_telemetry(self, protocol):
         if datetime.utcnow() >= (self._last_telemetry_heartbeat + UpdateHandler.TELEMETRY_HEARTBEAT_PERIOD):
             dropped_packets = self.osutil.get_firewall_dropped_packets(protocol.get_endpoint())
             auto_update_enabled = 1 if conf.get_autoupdate_enabled() else 0
-            # Include vm architecture in the heartbeat message because the kusto table does not have
-            # a separate column for it.
-            vmarch = self._get_vm_arch()
```
Comment (narrieta) [inline azurelinuxagent/ga/update.py:1018]: after removing this, self._get_vm_arch doesn't seem to be used anymore

Comment (maddieford) [reply azurelinuxagent/ga/update.py:1018]: It's used in the run method on line 318

File: azurelinuxagent/ga/update.py (hunk 2 above)
Comment (narrieta) [reply azurelinuxagent/ga/update.py:319]: ah, i see, this was intentional, following the lead of OS & python. OK as is, I guess.

**Review by narrieta — APPROVED** [review]

File: azurelinuxagent/common/event.py (hunk 1 above)
Comment (maddieford) [reply azurelinuxagent/common/event.py]: Updated to catch and return 'unknown' on failure

File: azurelinuxagent/ga/update.py (hunk 3 above)
Comment (maddieford) [reply azurelinuxagent/ga/update.py:1018]: I moved it to osutil since we are already using it in update and event

**Review by nagworld9 — APPROVED** [review]

**Review by narrieta — APPROVED** [review]

**Pull request merged**